feat(website): One tab for all protein sequences

theosanderson opened the pull request:
<!-- Add "preview" label in almost all cases to have testable deployment. Then lookup the deployment URL in argocd and paste it here (if you don't know how to look up the URL, ask here: https://loculus.slack.com/archives/C06JCAZLG14), it's something like `{REPLACE}.loculus.org` -->
preview URL: https://seq-changes.loculus.org/ebola-zaire/search?

- Fixes the height of the sequence container avoiding layout shifts when moving between tabs
- Uses a single tab for all proteins - selected with a dropdown- , rather than one tab per protein, which would be important for users working on some organisms
- Adjusts the names of the raw and aligned nucleotide tabs

<img width="1156" alt="image" src="https://github.com/user-attachments/assets/1615f3d0-543b-4d8f-b178-dee1ae4b63b8">

corneliusroemer reviewed (approved):
Really cool! Amazing that you improve the layout shift, has been something I'd noticed in the past.

There's still shift when switching to the aligned seqs because of the drop-down showing up.

In particular for multi-segment viruses the picker would make sense for all tabs. Not just aligned.

That would fix it. For non-segmented not sure.

chaoran-chen reviewed (changes requested):
Thanks Theo, this looks mostly really good to me!

One bug (I think it's a bug?) I noticed is that when you switch to the "Amino acid sequences" tab, you first still see the sequence from the tab that you have been previously on and no protein is selected. Only after selecting a protein, it shows an amino acid sequece.

![image](https://github.com/user-attachments/assets/dae6d646-98d7-43af-b67a-7f2b00790504)

chaoran-chen commented on website/src/components/SequenceDetailsPage/SequencesContainer.tsx:
```diff
@@ -154,15 +205,47 @@ const AlignmentSequenceTabs: FC<NucleotideSequenceTabsProps> = ({ nucleotideSegm
+                onChange={(e) => setType(geneSequence(e.target.value))}
+            >
+                <option value='' disabled>
+                    Select a protein
```
I think that we should avoid using the term "protein" here as not necessarily each AA seqeunce represents a different protein? Everywhere else, I believe that we have also rather just used "amino acid sequence" as term.

```suggestion
                    Select a sequence
```

theosanderson replied:
Hmm - you mean in that one protein can have multiple chains? In code we call these identifiers `genes` and I'd be OK with that here. I find `Select a sequence` a bit unexpected - here I'd expect the drop downs to be accessions. Would `Select a gene` work for you?

chaoran-chen replied:
For now, I think it's fine. When we look at more [schema designs](https://loculus.org/for-administrators/schema-designs/#multiple-references-for-an-organism) such as the multiple references for an organism approach, different AA sequences could be from the same gene aligned to different references.

theosanderson replied:
Sounds good to discuss then (I think in such a case I'd favour an approach where the UX involves selecting a reference and a gene(

corneliusroemer commented:
> One bug (I think it's a bug?)

Yeah you're right, I can now also reproduce and agree it's a bug

theosanderson commented:
Thanks - I fixed the bug by not displaying that old, wrong, sequence. I think the chances that users happen to want the top protein in the list are relatively low so I think this behaviour is OK for now - though I'm not averse to future PRs that change it. I adjusted `Select a protein` to `Select a gene`.

chaoran-chen commented on website/src/components/SequenceDetailsPage/SequencesContainer.tsx:
```diff
@@ -49,75 +49,118 @@ export const InnerSequencesContainer: FC<SequenceContainerProps> = ({
+                />
+                <BoxWithTabsTab
+                    isActive={activeTab === 'gene'}
+                    label='Amino acid sequences'
```
Shoud we rather name it "Aligned amino acid sequences" which would be more accurate? It might be worth emphasizing this so that users know that insertions are not included.

chaoran-chen reviewed (approved):
Thanks!
